### DISCUSSION TITLE
[SOLVED] Running Nextcloud AIO V3.2.0 behing Swag reverse proxy

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#1530] [Questions]
Hello,

I've search on other threads, but while the error message is the same, they don't seem using Swag reverse proxy, I'm using the standard Nextcloud behind Swag reverse proxy using Docker Compose and wanted to give Nextcloud AIO a try using the same, I've managed to get the docker compose running but when I get to the domain verification of the ne AIO instance I get the below error message:

Domain does not point to this server or the reverse proxy is not configured correctly.

and in the logs I have the below:

`Trying to fix docker.sock permissions internally...
Creating docker group internally with id 998
WARNING: No swap limit support
Initial startup of Nextcloud All In One complete!
You should be able to open the Nextcloud AIO Interface now on port 8080 of this server!
E.g. https://internal.ip.of.this.server:8080

If your server has port 80 and 8443 open and you point a domain to your server, you can get a valid certificate automatically by opening the Nextcloud AIO Interface via:
https://your-domain-that-points-to-this-server.tld:8443
{"level":"info","ts":1671145354.5627036,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
{"level":"warn","ts":1671145354.5642889,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/Caddyfile","line":2}
{"level":"info","ts":1671145354.5654352,"logger":"admin","msg":"admin endpoint started","address":"localhost:2019","enforce_origin":false,"origins":["//[::1]:2019","//127.0.0.1:2019","//localhost:2019"]}
{"level":"warn","ts":1671145354.566067,"logger":"http","msg":"server is listening only on the HTTP port, so no automatic HTTPS will be applied to this server","server_name":"srv0","http_port":80}
{"level":"warn","ts":1671145354.5663202,"logger":"http","msg":"automatic HTTP->HTTPS redirects are disabled","server_name":"srv1"}
{"level":"info","ts":1671145354.566293,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc0007722a0"}
{"level":"warn","ts":1671145354.566684,"logger":"tls","msg":"YOUR SERVER MAY BE VULNERABLE TO ABUSE: on-demand TLS is enabled, but no protections are in place","docs":"https://caddyserver.com/docs/automatic-https#on-demand-tls"}
{"level":"info","ts":1671145354.5669913,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/mnt/docker-aio-config/caddy/"}
{"level":"info","ts":1671145354.5672264,"logger":"tls","msg":"finished cleaning storage units"}
{"level":"info","ts":1671145354.5675745,"logger":"http.log","msg":"server running","name":"srv0","protocols":["h1","h2","h3"]}
{"level":"info","ts":1671145354.5676942,"logger":"http","msg":"enabling HTTP/3 listener","addr":":8443"}
{"level":"info","ts":1671145354.567998,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details."}
{"level":"info","ts":1671145354.568163,"logger":"http.log","msg":"server running","name":"srv1","protocols":["h1","h2","h3"]}
{"level":"error","ts":1671145354.568203,"msg":"unable to create folder for config autosave","dir":"/root/.config/caddy","error":"mkdir /root/.config: permission denied"}
{"level":"info","ts":1671145354.5682132,"msg":"serving initial configuration"}
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.23.0.2. Set the 'ServerName' directive globally to suppress this message
[Thu Dec 15 23:02:34.576256 2022] [ssl:warn] [pid 113] AH01906: 172.23.0.2:8080:0 server certificate is a CA certificate (BasicConstraints: CA == TRUE !?)
[Thu Dec 15 23:02:34.576430 2022] [ssl:warn] [pid 113] AH01909: 172.23.0.2:8080:0 server certificate does NOT include an ID which matches the server name
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.23.0.2. Set the 'ServerName' directive globally to suppress this message
[Thu Dec 15 23:02:34.596980 2022] [ssl:warn] [pid 113] AH01906: 172.23.0.2:8080:0 server certificate is a CA certificate (BasicConstraints: CA == TRUE !?)
[Thu Dec 15 23:02:34.597208 2022] [ssl:warn] [pid 113] AH01909: 172.23.0.2:8080:0 server certificate does NOT include an ID which matches the server name
[Thu Dec 15 23:02:34.600831 2022] [mpm_prefork:notice] [pid 113] AH00163: Apache/2.4.54 (Debian) PHP/8.0.26 OpenSSL/1.1.1n configured -- resuming normal operations
[Thu Dec 15 23:02:34.601015 2022] [core:notice] [pid 113] AH00094: Command line: 'apache2 -D FOREGROUND'
[Thu Dec 15 23:02:42.862185 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The response of the connection attempt to "https://mync.vortextech.fr:443" was: <html>    <head>        <title>Welcome to your SWAG instance</title>        <style>        body{            font-family: Helvetica, Arial, sans-serif;        }        .message{            width:440px;            padding:20px 40px;            margin:0 auto;            background-color:#f9f9f9;            border:1px solid #ddd;            color: #1e3d62;        }        center{            margin:40px 0;        }        h1{            font-size: 18px;            line-height: 26px;        }        p{            font-size: 12px;        }        a{            color: rgb(207, 48, 139);        }        </style>    </head>    <body>        <div class="message">            <h1>Welcome to your <a target="_blank" href="https://github.com/linuxserver/docker-swag">SWAG</a> instance</h1>            <p>A webserver and reverse proxy solution brought to you by <a target="_blank" href="https://www.linuxserver.io/">linuxserver.io</a> with php support and a built-in Certbot client.</p>            <p>We have an article on how to use swag here: <a target="_blank" href="https://docs.linuxserver.io/general/swag">docs.linuxserver.io</a></p>            <p>For help and support, please visit: <a target="_blank" href="https://www.linuxserver.io/support">linuxserver.io/support</a></p>        </div>    </body></html>, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:42.862440 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] Expected was: 7672426f9d29058fa5dba466c15c5737dec1b5b3823cfe02, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:42.862537 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The error message was: , referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:44.326445 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The response of the connection attempt to "https://mync.vortextech.fr:443" was: <html>    <head>        <title>Welcome to your SWAG instance</title>        <style>        body{            font-family: Helvetica, Arial, sans-serif;        }        .message{            width:440px;            padding:20px 40px;            margin:0 auto;            background-color:#f9f9f9;            border:1px solid #ddd;            color: #1e3d62;        }        center{            margin:40px 0;        }        h1{            font-size: 18px;            line-height: 26px;        }        p{            font-size: 12px;        }        a{            color: rgb(207, 48, 139);        }        </style>    </head>    <body>        <div class="message">            <h1>Welcome to your <a target="_blank" href="https://github.com/linuxserver/docker-swag">SWAG</a> instance</h1>            <p>A webserver and reverse proxy solution brought to you by <a target="_blank" href="https://www.linuxserver.io/">linuxserver.io</a> with php support and a built-in Certbot client.</p>            <p>We have an article on how to use swag here: <a target="_blank" href="https://docs.linuxserver.io/general/swag">docs.linuxserver.io</a></p>            <p>For help and support, please visit: <a target="_blank" href="https://www.linuxserver.io/support">linuxserver.io/support</a></p>        </div>    </body></html>, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:44.326709 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] Expected was: 7672426f9d29058fa5dba466c15c5737dec1b5b3823cfe02, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:44.326820 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The error message was: , referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:45.407799 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The response of the connection attempt to "https://mync.vortextech.fr:443" was: <html>    <head>        <title>Welcome to your SWAG instance</title>        <style>        body{            font-family: Helvetica, Arial, sans-serif;        }        .message{            width:440px;            padding:20px 40px;            margin:0 auto;            background-color:#f9f9f9;            border:1px solid #ddd;            color: #1e3d62;        }        center{            margin:40px 0;        }        h1{            font-size: 18px;            line-height: 26px;        }        p{            font-size: 12px;        }        a{            color: rgb(207, 48, 139);        }        </style>    </head>    <body>        <div class="message">            <h1>Welcome to your <a target="_blank" href="https://github.com/linuxserver/docker-swag">SWAG</a> instance</h1>            <p>A webserver and reverse proxy solution brought to you by <a target="_blank" href="https://www.linuxserver.io/">linuxserver.io</a> with php support and a built-in Certbot client.</p>            <p>We have an article on how to use swag here: <a target="_blank" href="https://docs.linuxserver.io/general/swag">docs.linuxserver.io</a></p>            <p>For help and support, please visit: <a target="_blank" href="https://www.linuxserver.io/support">linuxserver.io/support</a></p>        </div>    </body></html>, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:45.408076 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] Expected was: 7672426f9d29058fa5dba466c15c5737dec1b5b3823cfe02, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:45.408201 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The error message was: , referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:46.387735 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The response of the connection attempt to "https://mync.vortextech.fr:443" was: <html>    <head>        <title>Welcome to your SWAG instance</title>        <style>        body{            font-family: Helvetica, Arial, sans-serif;        }        .message{            width:440px;            padding:20px 40px;            margin:0 auto;            background-color:#f9f9f9;            border:1px solid #ddd;            color: #1e3d62;        }        center{            margin:40px 0;        }        h1{            font-size: 18px;            line-height: 26px;        }        p{            font-size: 12px;        }        a{            color: rgb(207, 48, 139);        }        </style>    </head>    <body>        <div class="message">            <h1>Welcome to your <a target="_blank" href="https://github.com/linuxserver/docker-swag">SWAG</a> instance</h1>            <p>A webserver and reverse proxy solution brought to you by <a target="_blank" href="https://www.linuxserver.io/">linuxserver.io</a> with php support and a built-in Certbot client.</p>            <p>We have an article on how to use swag here: <a target="_blank" href="https://docs.linuxserver.io/general/swag">docs.linuxserver.io</a></p>            <p>For help and support, please visit: <a target="_blank" href="https://www.linuxserver.io/support">linuxserver.io/support</a></p>        </div>    </body></html>, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:46.388006 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] Expected was: 7672426f9d29058fa5dba466c15c5737dec1b5b3823cfe02, referer: https://10.0.1.153:8195/containers
[Thu Dec 15 23:02:46.388108 2022] [php:notice] [pid 145] [client 127.0.0.1:42628] The error message was: , referer: https://10.0.1.153:8195/containers

`

not sure what I've setup incorrectly and I'm most probably missing something obvious here, but can't see what; if anyone has managed to get this working with Swag and could point me in the right direction, I would be very grateful.

Thanks in advance

Al

**Comment by szaimen** [comment]
Hi, did you see https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md?

**Reply by ghost** [reply]
Hi, Yes I have but unfortunately, it's not clear to me which configuration example is applicable in my case using the  swag reverse proxy. 

I'll like to get this running te normal way but if worse comes to worst, can I skip the domain validation when using docker compose?

**Comment by szaimen** [comment]
Can you share your docker-compose or docker run command of AIO and Swag?

**Reply by ghost** [reply]
sure, here you go:

Nextcloud AIO:

`
---
version: "3.8"
volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
  Caddyfile:  
services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: always
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 8195:8080
    environment:
      - APACHE_PORT=11000
      - TRUSTED_PROXIES=10.0.1.150
      - NEXTCLOUD_TRUSTED_DOMAINS=mync.vortextech.fr 10.0.1.150`

Swag:

`---
version: "3.8"

secrets:
  JWT_SECRET:
    file: ./data/authelia/secrets/JWT_SECRET
  STORAGE_ENCRYPTION_KEY:
    file: ./data/authelia/secrets/STORAGE_ENCRYPTION_KEY
volumes:
  swag:
  authelia:

networks:
  net:
    driver: bridge

services:
  swag:
    container_name: swag
    image: lscr.io/linuxserver/swag
    restart: unless-stopped
    networks:
      net:
        aliases: []
    ports:
      - 80:80
      - 443:443
      - 81:81
    volumes:
      - ./data/swag:/config
    environment:
      PUID: 1000
      PGID: 1000
      TZ: Europe/Paris
      URL: vortextech.fr
      SUBDOMAINS: wildcard
      VALIDATION: dns
      CERTPROVIDER: cloudflare
      DNSPLUGIN: cloudflare
      DOCKER_MODS: linuxserver/mods:swag-dashboard|linuxserver/mods:swag-mod2
      EMAIL: mymail@mail.com
    cap_add:
      - NET_ADMIN
  authelia:
    container_name: authelia
    image: authelia/authelia
    restart: unless-stopped
    networks:
      net:
        aliases: []
    expose:
      - 9091
    secrets: [JWT_SECRET, SESSION_SECRET, STORAGE_PASSWORD, STORAGE_ENCRYPTION_KEY]  
    volumes:
      - ./data/authelia/config:/config
    environment:
      TZ: Europe/Paris
      AUTHELIA_JWT_SECRET_FILE: /run/secrets/JWT_SECRET
      AUTHELIA_STORAGE_ENCRYPTION_KEY_FILE: /run/secrets/STORAGE_ENCRYPTION_KEY
      AUTHELIA_SESSION_NAME: authelia_session
      AUTHELIA_SESSION_DOMAIN: vortextech.fr
      AUTHELIA_SESSION_SAME_SITE: lax
      AUTHELIA_SESSION_EXPIRATION: 1h
      AUTHELIA_SESSION_INACTIVITY: 5m
      AUTHELIA_STORAGE_LOCAL_PATH: /config/db.sqlite3
      AUTHELIA_NOTIFIER_FILESYSTEM_FILENAME: /config/notification.txt
      AUTHELIA_AUTHENTICATION_BACKEND_FILE_PATH: /config/users_database.yml
      AUTHELIA_THEME: dark
      AUTHELIA_ACCESS_CONTROL_DEFAULT_POLICY: deny`

**Comment by szaimen** [comment]
Basically you still need to follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md. IIRC is SWAG based on Nginx so you should be able to use that template.

**Reply by ghost** [reply]
Hi Szaimen,

Thanks for the response, so to make sure I get this right do I need to use the Nginx template, ok but should I replace

this:

```
server {
    listen 11000 ssl;
    listen [::]:443 ssl;

    server_name mync.*;

    include /config/nginx/ssl.conf;

    client_max_body_size 0;

    location / {
        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
        set $upstream_app 10.0.1.153;
        set $upstream_port 8195;
        set $upstream_proto https;
        proxy_pass $upstream_proto://$upstream_app:$upstream_port;

        proxy_hide_header X-Frame-Options;
        proxy_max_temp_file_size 2048m;
    }
}
```

with this:

```
map $http_upgrade $connection_upgrade {
    default upgrade;
    '' close;
}

server {
    listen 80;
    listen [::]:80;            # comment to disable IPv6

    if ($scheme = "http") {
        return 301 https://$host$request_uri;
    }

    listen 443 ssl http2;
    listen [::]:443 ssl http2; # comment to disable IPv6

    server_name <your-nc-domain>;

    location / {
        resolver localhost;
        proxy_pass http://localhost:11000$request_uri;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        client_max_body_size 0;

        # Websocket
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
    }

    ssl_certificate /etc/letsencrypt/live/<your-nc-domain>/fullchain.pem;   # managed by certbot on host machine
    ssl_certificate_key /etc/letsencrypt/live/<your-nc-domain>/privkey.pem; # managed by certbot on host machine

    ssl_session_timeout 1d;
    ssl_session_cache shared:MozSSL:10m; # about 40000 sessions
    ssl_session_tickets off;

    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
    ssl_prefer_server_ciphers off;
}
```

or is this to be added to the original template; also FWIW do note that the proxy is not running on the same machine, so not sure if that makes any difference.

Thanks again

**Reply by szaimen** [reply]
Yes, please use the suggested template.

If the reverse proxy is not running on the same machine, you need to substitute `localhost` in the template through the private ip-address of the server where AIO is running on as it is described in the documentation.

**Reply by szaimen** [reply]
btw, if SWAG hanlded certificates automatically, you likely don't need to specify the location of the certificate...

**Reply by ghost** [reply]
Thanks, I've done as suggested and manage to get some progress done and still got an error message, but a different one this time as per below: 

The server is not reachable on Port 443. You can verify this e.g. with 'https://portchecker.co/' by entering your domain there as ip-address and port 443 as port.

here's what I've replace the original template with:

` 
server {
    listen 80;
    listen [::]:80;            # comment to disable IPv6

    if ($scheme = "http") {
        return 301 https://$host$request_uri;
    }

    listen 443 ssl http2;
    listen [::]:443 ssl http2; # comment to disable IPv6

    server_name mync.vortextech.fr;

    location / {
        resolver localhost;
        proxy_pass http://10.0.1.153:11000$request_uri;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        client_max_body_size 0;

        # Websocket
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
    }

    ssl_certificate /etc/letsencrypt/live/mync.vortextech.fr/fullchain.pem;   # managed by certbot on host machine
    ssl_certificate_key /etc/letsencrypt/live/mync.vortextech.fr/privkey.pem; # managed by certbot on host machine

    ssl_session_timeout 1d;
    ssl_session_cache shared:MozSSL:10m; # about 40000 sessions
    ssl_session_tickets off;

    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
    ssl_prefer_server_ciphers off;
}
`

I'm assuming this is about port 443 on the host where Nextcloud lives as the port is open on my firewall and points to swag reverse proxy

**Reply by ghost** [reply]
> btw, if SWAG hanlded certificates automatically, you likely don't need to specify the location of the certificate...

you talking about this 2 lines ?

`ssl_certificate /etc/letsencrypt/live/mync.vortextech.fr/fullchain.pem;   # managed by certbot on host machine
ssl_certificate_key /etc/letsencrypt/live/mync.vortextech.fr/privkey.pem; # managed by certbot on host machine
`

**Reply by szaimen** [reply]
Yes, however after thinking about it you probably need it...

**Reply by szaimen** [reply]
btw, can you check with the portchecker if the port is open? If not, you need to fix your dns and domain config.

**Reply by ghost** [reply]
> btw, can you check with the portchecker if the port is open? If not, you need to fix your dns and domain config.

I've checked with portchecker and it states the port is closed which is weird as I have the port open and forwarded on the firewall to point to the reverse proxy, I'll need to make some more digging on that front and get back to Nextcloud once that's sorted.

thanks again for you precious help so far

**Comment by ghost** [comment]
Hello, managed to get everything working, the domain related to port 443 not being accessible were due to the fact that for an unknown reason there was a glitch on my internet connection, managed to get it solved by just rebooting my router and the fiber optic interface from my ISP; following, it all worked like a charm! 

Thanks again for the help with all the other errors I got while setting this up.

**Reply by ThePhaseless** [reply]
can you share your docker-compose file once again? i dont know which one is working. Also, 10.0.1.153 is ip of nextcloud or swag?

**Reply by ghost** [reply]
Hello, 

The Docker is already posted several posts above on this thread, 10.0.1.153 is Nextcloud AIO, the Proxy is on a different ip; what exactly are you after?

**Reply by serch826** [reply]
Hello, what port number should put in the swag nextcloud-aio subdomain config?
